imap-proto/src/parser/gmail.rs

```diff
@@ -0,0 +1,52 @@
+use std::borrow::Cow;
+
+use nom::branch::alt;
+use nom::bytes::streaming::tag_no_case;
+use nom::combinator::map;
+use nom::sequence::preceded;
+use nom::IResult;
+
+use crate::{AttributeValue, MailboxDatum};
+
+use super::core::{parenthesized_list, quoted_utf8};
+use super::rfc3501::flag;
+
+pub(crate) fn x_gm_label_list(i: &[u8]) -> IResult<&[u8], Vec<Cow<str>>> {
```

Comment:
Rather than using the `x_gm_` prefix on these functions, let's just replace that with `gmail`, since that seems the intended meaning.